hw/mcu/nordic/nrf52xxx/src/nrf52_clock.c

```diff
@@ -44,7 +44,18 @@ nrf52_clock_hfxo_request(void)
     __HAL_DISABLE_INTERRUPTS(ctx);
     assert(nrf52_clock_hfxo_refcnt < 0xff);
     if (nrf52_clock_hfxo_refcnt == 0) {
-        NRF_CLOCK->TASKS_HFCLKSTART = 1;
+        /* Check the current STATE and SRC of HFCLK */
+        if ((NRF_CLOCK->HFCLKSTAT &
+             (CLOCK_HFCLKSTAT_SRC_Msk | CLOCK_HFCLKSTAT_STATE_Msk)) !=
+            (CLOCK_HFCLKSTAT_SRC_Msk | CLOCK_HFCLKSTAT_STATE_Msk)) {
+            NRF_CLOCK->EVENTS_HFCLKSTARTED = 0;
+            NRF_CLOCK->TASKS_HFCLKSTART = 1;
+            while (1) {
+                if ((NRF_CLOCK->EVENTS_HFCLKSTARTED) != 0) {
+                    break;
+                }
```

Comment:
what about using this here?
```
while (!NRF_CLOCK->EVENTS_HFCLKSTARTED) {
}
```

Reply:
Again, same reason - minimum changes left this one in there. Very happy to simplify this.